miniapps/tribol/ContactPatchTest.cpp

```diff
@@ -0,0 +1,327 @@
+   std::set<int> mortar_attrs({4});
+   // z=0.99 plane of top block
+   std::set<int> nonmortar_attrs({5});
+   // per-dimension sets of boundary attributes with homogeneous Dirichlet BCs
+   std::vector<std::set<int>> fixed_attrs(3);
+   fixed_attrs[0] = {1}; // x=0 plane of both blocks
+   fixed_attrs[1] = {2}; // y=0 plane of both blocks
+   fixed_attrs[2] = {3, 6}; // 3: z=0 plane of bottom block; 6: z=1.99 plane of top block
+
+   // Read the mesh, refine, and create a mfem::ParMesh
+   mfem::Mesh serial_mesh(mesh_file);
+   for (int i = 0; i < ref_levels; ++i)
+   {
+      serial_mesh.UniformRefinement();
+   }
+   mfem::ParMesh mesh(MPI_COMM_WORLD, serial_mesh);
+   serial_mesh.Clear();
+
+   // Create an H1 finite element space on the mesh for displacements/forces
+   mfem::H1_FECollection fec(order, 3);
+   mfem::ParFiniteElementSpace fespace(&mesh, &fec, 3);
+   auto n_displacement_dofs = fespace.GlobalTrueVSize();
+   if (mfem::Mpi::Root())
+   {
+      std::cout << "Number of displacement unknowns: " << n_displacement_dofs <<
+                std::endl;
+   }
+
+   // Create coordinate and displacement grid functions
+   mfem::ParGridFunction coords(&fespace);
+   mesh.SetNodalGridFunction(&coords);
+   mfem::ParGridFunction displacement(&fespace);
+   displacement = 0.0;
+
+   // Find true dofs with homogeneous Dirichlet BCs
+   mfem::Array<int> ess_tdof_list;
+   {
+      mfem::Array<int> ess_vdof_marker(fespace.GetVSize());
+      ess_vdof_marker = 0;
+      for (int i = 0; i < 3; ++i)
+      {
+         mfem::Array<int> ess_bdr(mesh.bdr_attributes.Max());
+         ess_bdr = 0;
+         for (auto xfixed_attr : fixed_attrs[i])
+         {
+            ess_bdr[xfixed_attr-1] = 1;
+         }
+         mfem::Array<int> new_ess_vdof_marker;
+         fespace.GetEssentialVDofs(ess_bdr, new_ess_vdof_marker, i);
+         for (int j = 0; j < new_ess_vdof_marker.Size(); ++j)
+         {
+            ess_vdof_marker[j] = ess_vdof_marker[j] || new_ess_vdof_marker[j];
+         }
+      }
+      mfem::Array<int> ess_tdof_marker;
+      fespace.GetRestrictionMatrix()->BooleanMult(ess_vdof_marker, ess_tdof_marker);
+      mfem::FiniteElementSpace::MarkerToList(ess_tdof_marker, ess_tdof_list);
+   }
+
+   // Create small deformation linear elastic bilinear form
+   mfem::ParBilinearForm a(&fespace);
+   mfem::ConstantCoefficient lambda_coeff(lambda);
+   mfem::ConstantCoefficient mu_coeff(mu);
+   a.AddDomainIntegrator(new mfem::ElasticityIntegrator(lambda_coeff, mu_coeff));
+
+   // Compute elasticity contribution to tangent stiffness matrix
+   a.Assemble();
+   std::unique_ptr<mfem::HypreParMatrix> A(new mfem::HypreParMatrix);
+   a.FormSystemMatrix(ess_tdof_list, *A);
+
+   // Initialize Tribol contact library
+   tribol::initialize(3, MPI_COMM_WORLD);
+
+   // Create a Tribol coupling scheme: defines contact surfaces and enforcement
+   int coupling_scheme_id = 0;
+   int mesh1_id = 0;
+   int mesh2_id = 1;
```

Comment:
Minor: consider clarifying that while there is a single mfem mesh, Tribol requires unique mesh ids for the mortar method, essentially referring to the contact surface ids.